pkg/controller/operators/catalog/operator.go

```diff
@@ -976,22 +976,28 @@ func (o *Operator) syncResolvingNamespace(obj interface{}) error {
 		// not-satisfiable error
 		if _, ok := err.(solver.NotSatisfiable); ok {
 			logger.WithError(err).Debug("resolution failed")
-			updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ConstraintsNotSatisfiable", err.Error(), true)
+			_, updateErr := o.updateSubscriptionStatuses(subs)
 			if updateErr != nil {
 				logger.WithError(updateErr).Debug("failed to update subs conditions")
+				return updateErr
 			}
 			return nil
 		}
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "ErrorPreventedResolution", err.Error(), true)
+		_, updateErr := o.updateSubscriptionStatuses(subs)
 		if updateErr != nil {
 			logger.WithError(updateErr).Debug("failed to update subs conditions")
+			return updateErr
 		}
 		return err
 	} else {
-		updateErr := o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
+		subs = o.setSubsCond(subs, v1alpha1.SubscriptionResolutionFailed, "", "", false)
+		_, updateErr := o.updateSubscriptionStatuses(subs)
 		if updateErr != nil {
 			logger.WithError(updateErr).Debug("failed to update subs conditions")
```

Comment:
This one might not be visible as a sync error, and it might mean that there is a Subscription with a misleading status, so logging it at warning seems useful to me:

```suggestion
			logger.WithError(updateErr).Warn("failed to update subscription conditions")
```